### ISSUE TITLE
Resend ACTION-VIEW with DATA-https-URL does not work on android-7

### ISSUE THREAD

**Issue by TPS** [k3b/intent-intercept#4]
@k3b I think this is https://github.com/intrications/intent-intercept/issues/42, but another instance:

When using Ff Klar/Focus (from https://f-droid.org/app/org.mozilla.klar, https://play.google.com/store/apps/details?id=org.mozilla.klar, or https://play.google.com/store/apps/details?id=org.mozilla.focus), if 1 uses "Open with…" from the ⋮ menu, it works perfectly when 1 chooses another browser, but if 1 chooses II (e.g., if 1 wants to edit the URL before opening elseplace), II's "Send Edited Intent" remains faded out, no matter whatever else I do.  (The same bug is in the original II.)

Is there any other info I can provide?

_OT:  Are you interested in continuing to fix issues from https://github.com/intrications/intent-intercept/issues?_

**Comment by k3b** [comment]
is the send button also faded out if you do not edit the url?

I assume that "faded out" means that no app matches the intent possible reasons:

* IntentIntercept receives some data that 
  * II cannot resend or 
  * that II receives as a temporary open http(s):-stream but that  II can only retransmit as a string
* some permission issue

Firefox does not work on my android-4.4 because it has not enough cpu-memory so i cannot test it myself.

please add to this ticket:

* used android-versoin (mine are currently "4.4 rooted, non-google" and "4.2 non-rooted") 
* the content of the intercepted intent as text (via the send/share button in the upper right corner) so i can further analyse it

**Comment by TPS** [comment]
- "Send" button starts faded-out in this instance
- Ff __Klar/Focus__ are Webview/Chrome (not Gecko) engine, so should be comparatively lightweight enough to run on 4.×.  (I'd similar problems when I'd devices w/ 4.4.2 & 4.4.4, so I understand completely.)
- I'm currently on Android 7.1.1.
- For the URL for your reply (above), text content as follows
````
intent://github.com/k3b/intent-intercept/issues/4#issuecomment-352891617#Intent;scheme=https;package=de.k3b.android.intentintercept;end 
------------ 
ACTION: android.intent.action.VIEW 
DATA: https://github.com/k3b/intent-intercept/issues/4#issuecomment-352891617 
URI: intent://github.com/k3b/intent-intercept/issues/4#issuecomment-352891617#Intent;scheme=https;package=de.k3b.android.intentintercept;end 
FLAGS: 
none 

------------ 
MATCHING ACTIVITIES: 
none 
````

**Comment by k3b** [comment]
Interesting. This seems to be a android-7 permission issue because of `FLAGS: none`

Under android-4.4 i have FLAGS: FLAG_ACTIVITY_FORWARD_RESULT

If you add 
<pre>launchFlags=0x13104000;</pre> to the intent to manually set the FLAGS similar to my example below.
Will this allow the re-send ?

---
This is a similar intent from my android-4.4 device where re-send works for view https:....

It was generated from a app-launcher-bookmark from privacybrowser (com.stoutner.privacybrowser.standard)
 
<pre>
intent://m.heise.de/newsticker/#Intent;scheme=https;action=android.intent.action.VIEW;launchFlags=0x13104000;sourceBounds=198%20630%20444%20848;end 
------------ 
ACTION: android.intent.action.VIEW 
DATA: https://m.heise.de/newsticker/ 
URI: intent://m.heise.de/newsticker/#Intent;scheme=https;action=android.intent.action.VIEW;launchFlags=0x13104000;sourceBounds=198%20630%20444%20848;end 
FLAGS: 
FLAG_ACTIVITY_FORWARD_RESULT 
FLAG_ACTIVITY_TASK_ON_HOME 
FLAG_ACTIVITY_PREVIOUS_IS_TOP 
FLAG_RECEIVER_FOREGROUND 
FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY 

------------ 
MATCHING ACTIVITIES: 
Browser (com.android.browser - com.android.browser.BrowserActivity) 
Chrome (com.android.chrome - com.google.android.apps.chrome.Main) 
Opera Mobile (com.opera.browser - com.opera.Opera) 
Privacy Browser (com.stoutner.privacybrowser.standard - com.stoutner.privacybrowser.activities.MainWebViewActivity) 
Intent Abfänger (de.k3b.android.intentintercept - uk.co.ashtonbrsc.intentexplode.Explode) 
==> Zip (de.k3b.android.toGoZip - de.k3b.android.toGoZip.Add2ZipActivity) 

</pre>

**Comment by k3b** [comment]
No solution yet.

what i tried on android-7.1 using firefor-klar

* temporarly added internet-permissions : 
* replaced parameters Intent.toUri/Intent.parseUri from Intent.URI_INTENT_SCHEME to Intent.URI_INTENT_SCHEME | URI_ALLOW_UNSAFE

no change :-(

**Comment by TPS** [comment]
Sorry for delay replying. I tried adding `launchFlags=0x13104000` as mentioned. That populated `FLAGS`, but didn't cause any apps to be matched, as you said above.

**Comment by TPS** [comment]
@k3b What's interesting about this is that I _don't_ replicate this w/ other browsers that _also_ have an "Open with…" option, e.g., http://play.google.com/store/apps/details?id=com.cleanmaster.security, which has a Private Browsing option. When I use it to open II from this page, I get
````
intent://github.com/k3b/intent-intercept/issues/4#Intent;scheme=https;end 
------------ 
ACTION: android.intent.action.VIEW 
DATA: https://github.com/k3b/intent-intercept/issues/4 
URI: intent://github.com/k3b/intent-intercept/issues/4#Intent;scheme=https;end 
FLAGS: 
none 

------------ 
MATCHING ACTIVITIES: 
Safe Browsing (com.cleanmaster.security - ks.cm.antivirus.privatebrowsing.PrivateBrowsingActivity) 
Share URL (com.intrications.android.sharebrowser - com.intrications.android.sharebrowser.MainActivity) 
LastPass (com.lastpass.lpandroid - com.lastpass.lpandroid.activity.WebBrowserActivity) 
LastPass Beta (com.lastpass.lpandroid.beta - com.lastpass.lpandroid.activity.WebBrowserActivity) 
Firefox Klar (org.mozilla.klar - org.mozilla.focus.activity.MainActivity)
````
which works _perfectly_.  I think the only difference I see is the `package=de.k3b.android.intentintercept;` portion of the URI/intent (that Klar has, but Private Browsing doesn't).  Removing it manually doesn't seem to help, nor replacing it w/ the `FLAGS` option above,  though.  Still, I suppose this might be considered a bug in Klar (in that it sends that), but I figure it's probably a common-enough issue (e.g., probably _all_ Ff browsers do this, if not other apps) that might be pretty easy to workaround, @ least as _part_ of a solution.

**Comment by k3b** [comment]
I assume that intentintercept needs android-6-style dynamic internet permission for android-7 and later.

Mergerequests are welcome

**Comment by TPS** [comment]
I think no, because whenever I try the same sequence w/ any other browser (I use LastPass & Edge, also), it just works.…

**Comment by k3b** [comment]
Fixed in Version 4.0.0
